Source/WarpX.cpp

```diff
@@ -1010,6 +1010,12 @@ WarpX::ReadParameters ()
             maxLevel() <= 0,
             "Vay deposition not implemented with mesh refinement");
 
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            !(WarpX::current_deposition_algo == CurrentDepositionAlgo::Vay and
+             (electromagnetic_solver_id == ElectromagneticSolverAlgo::Yee ||
+              electromagnetic_solver_id == ElectromagneticSolverAlgo::CKC)),
+            "Vay deposition not implemented with ckc or yee");
+
```

Comment:
Responding to your question, yes, Vay deposition is only implemented for the PSATD solver. I do not recall discussing its implementation for the FDTD case in the near future. The algorithm itself, as it is described in the literature ([Vay et al, JCP, 2013](https://doi.org/10.1016/j.jcp.2013.03.010), equations (20)-(24)), is designed for a spectral method, as the true current J is computed from the deposited current D through some operations in k-space.

Hence, I agree that we could add a condition `electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD` here. Personally, I find the conditions like the one implemented now, `!(a && (b || c))`, relatively intricated and not easy to decipher. I would be in favor of a more verbose but easier implementation, such as
```
if (WarpX::current_deposition_algo == CurrentDepositionAlgo::Vay)
{
    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
        electromagnetic_solver_id == ElectromagneticSolverAlgo::PSATD,
        "Vay deposition implemented only for algo.maxwell_solver=psatd");
}
```

Reply:
sounds good. I will add this and will also add that condition whereever needed then